src/vm/vm_memory/memory.rs

```diff
@@ -527,6 +541,82 @@ mod memory_tests {
         );
     }
 
+    #[test]
+    fn validate_existing_memory_for_invalid_signature() {
+        let mut builtin = SignatureBuiltinRunner::new(&EcdsaInstanceDef::default(), true);
+
+        let signing_key = SigningKey::random(&mut OsRng);
+
+        let verifying_key = VerifyingKey::from(&signing_key);
+
+        let mut segments = MemorySegmentManager::new();
+        let mut memory = Memory::new();
+        segments.add(&mut memory);
+        builtin.initialize_segments(&mut segments, &mut memory);
+        memory
+            .insert(
+                &MaybeRelocatable::from((1, 0)),
+                &MaybeRelocatable::from(BigInt::from_bytes_be(
+                    Sign::Plus,
+                    verifying_key.to_bytes().as_slice(),
+                )),
+            )
+            .unwrap();
+        memory
+            .insert(
+                &MaybeRelocatable::from((1, 1)),
+                &MaybeRelocatable::from(bigint_str!(b"-1472574760335685482768423018116732869320670550222259018541069375211356613248")),
+            ).unwrap();
+        builtin.add_validation_rule(&mut memory).unwrap();
+        let error = memory.validate_existing_memory();
+        assert!(error.is_err())
```

Comment:
Can't we use an assert_eq! and compare the error to a concrete error here?,  keeping the error info available is more useful when debugging